src/server/api/schema.js

```diff
@@ -622,6 +623,77 @@ const rootMutations = {
       const newCampaign = await campaignInstance.save()
       return editCampaign(newCampaign.id, campaign, loaders)
     },
+    copyCampaign: async (_, { id }, { user, loaders }) => {
+      const campaign = await loaders.campaign.load(id)
+      await accessRequired(user, campaign.organization_id, 'ADMIN')
+
+      const campaignInstance = new Campaign({
+        organization_id: campaign.organization_id,
+        title: 'COPY - ' + campaign.title,
+        description: campaign.description,
+        due_by: campaign.dueBy,
+        is_started: false,
+        is_archived: false
+      })
+      const newCampaign = await campaignInstance.save()
+      const newCampaignId = newCampaign.id
+      const oldCampaignId = campaign.id
+
+      let interactions = await r.knex('interaction_step')
+        .where({campaign_id: oldCampaignId })
+
+      const interactionsArr = []
+      interactions.forEach((interaction, index) => {
+        if(interaction.parent_interaction_id){
+          let is = {
+            id: 'new'+interaction.id,
+            questionText: interaction.question,
+            script: interaction.script,
+            answerOption: interaction.answer_option,
+            answerActions: interaction.answer_actions,
+            isDeleted: interaction.is_deleted,
+            campaign_id: newCampaignId,
+            parentInteractionId: 'new'+interaction.parent_interaction_id
+          }
+          interactionsArr.push(is)
+        } else if (!interaction.parent_interaction_id){
+          let is = {
+            id: 'new'+interaction.id,
+            questionText: interaction.question,
+            script: interaction.script,
+            answerOption: interaction.answer_option,
+            answerActions: interaction.answer_actions,
+            isDeleted: interaction.is_deleted,
+            campaign_id: newCampaignId,
+            parentInteractionId: interaction.parent_interaction_id
+          }
+          interactionsArr.push(is)
+        }
+      })
+
+      let createSteps = updateInteractionSteps(newCampaignId, [makeTree(interactionsArr, id = null)], campaign, {})
```

Comment:
Can we write this `const createSteps = await updateInteractionSteps(....)` and remove the line below?